test/models/shared/taxonomies_base_test.rb

```diff
@@ -0,0 +1,441 @@
+
+      # inherit parameter from parent
+      child_taxonomy = taxonomy_class.create :name => "floor1", :parent_id => taxonomies(:"#{taxonomy_name}1").id
+      parent_params = taxonomies(:"#{taxonomy_name}1").public_send("#{taxonomy_name}_parameters")
+      assert_equal [], child_taxonomy.public_send("#{taxonomy_name}_parameters")
+
+      # new parameter on child taxonomy
+      child_taxonomy.public_send("#{taxonomy_name}_parameters").create(:name => "child_param", :value => "123")
+
+      assert_equal({ tax_param => parent_params.first.value }, child_taxonomy.parent_params)
+    end
+
+    test "#params_objects should return taxonomy parameters" do
+      taxonomy = taxonomy_class.create :name => "floor1"
+      param = "#{taxonomy_name}Parameter".classify.constantize.
+        create(:name => 'name', :value => 'valueable')
+      taxonomy.public_send("#{taxonomy_name}_parameters=", [param])
+      assert(taxonomy.params_objects.include?(param))
+    end
+
+    test "#params_objects should return ancestors parameters" do
+      taxonomy = taxonomy_class.create :name => "floor1", :parent_id => taxonomies(:"#{taxonomy_name}1").id
+      assert_equal [], taxonomy.public_send("#{taxonomy_name}_parameters")
+      assert_equal [taxonomies(:"#{taxonomy_name}1").public_send("#{taxonomy_name}_parameters")], taxonomy.params_objects
+    end
+
+    test "cannot delete taxonomy that is a parent for nested taxonomy" do
+      parent1 = taxonomies(:"#{taxonomy_name}2")
+      taxonomy_class.create :name => "floor1", :parent_id => parent1.id
+      assert_raise Ancestry::AncestryException do
+        parent1.destroy
+      end
+    end
+
+    test "taxonomy name can't be too big to create lookup value matcher over 255 characters" do
+      parent = FactoryGirl.create(:"#{taxonomy_name}")
+      min_lookupvalue_length = "#{taxonomy_name}=".length + parent.title.length + 1
+      taxonomy = taxonomy_class.new :parent => parent, :name => 'a' * (256 - min_lookupvalue_length)
+      refute_valid taxonomy
+      assert_equal "is too long (maximum is %s characters)" % (255 - min_lookupvalue_length),
+        taxonomy.errors[:name].first
+    end
+
+    test "taxonomy name can be up to 255 characters" do
+      parent = FactoryGirl.create(:"#{taxonomy_name}")
+      min_lookupvalue_length = "#{taxonomy_name}=".length + parent.title.length + 1
+      taxonomy = taxonomy_class.new :parent => parent, :name => 'a' * (255 - min_lookupvalue_length)
+      assert_valid taxonomy
+    end
+
+    test "taxonomy should not save when matcher is exactly 256 characters" do
+      parent = FactoryGirl.create(:"#{taxonomy_name}", :name => 'a' * (255 - taxonomy_name.length - 2))
+      taxonomy = taxonomy_class.new :parent => parent, :name => 'b'
+      refute_valid taxonomy
+      assert_equal _("is too long (maximum is 0 characters)"), taxonomy.errors[:name].first
+    end
+
+    test 'ignores the taxable_type if current taxonomy ignores it' do
```

Comment:
These two are the only new tests, the rest of them are just removing the duplication between location_test and organization_test and moving them both to a module